### PR TITLE
Integration test cleanup

### PR DESCRIPTION
**Pull request by gzsombor**
Following the recent test cleanup patch, similarly, these converts the` <% if (something) %>someText<% } %> ` into `<%= variable %>` - which makes the code bit readable, and allows to customize the conditions a bit easier in the future.

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs [hunk 1]
```diff
@@ -628,13 +641,13 @@ public class <%= entityClass %>ResourceIT <% if (databaseType === 'cassandra') {
 
     <%_ if (databaseType === 'sql' && isUsingMapsId === true) { _%>
     @Test
-    @Transactional
+    <%= transactionalAnnotation %>
```
Comment (atomfrede) [inline generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs:644]: I think it should be on the same line here, as `transactionalAnnotation` contains the linebreak already.

File: generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs [hunk 2]
```diff
@@ -42,6 +42,21 @@ if (reactive) {
     callBlock = ".block()";
     callListBlock = ".collectList().block()";
 }
+let repositorySave = 'save';
```
Comment (atomfrede) [inline generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs:45]: I think naming it `saveMethod` (or just `save`) would make understanding the templates a little easier:
```
<%= entityInstance %>Repository.<%= saveMethod %>
<%= entityInstance %>Repository.<%= save %>
```

File: generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs (hunk 1 above)
Comment (gzsombor) [reply generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs]: Yes, you are right

File: generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs (hunk 2 above)
Comment (gzsombor) [reply generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs]: Done

File: generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs [hunk 3]
```diff
@@ -42,6 +42,21 @@ if (reactive) {
...
+let saveMethod = 'save';
+if (!reactive && databaseType === 'sql') {
+    saveMethod = 'saveAndFlush';
+}
+let idValue = `${asEntity(entityInstance)}.getId()`;
+if (databaseType === 'sql' && hasOauthUser !== true) {
+    idValue = idValue + '.intValue()';
+} else if (databaseType === 'cassandra') {
+    idValue = idValue + '.toString()';
+}
```
Comment (murdos) [inline generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs:54]: Could you replace these conditions by the following ones, that are more logical and direct (I know this was like that in previous code, but it's a good opportunity to improve it)
```suggestion
if (primaryKeyType === 'Long') {
    idValue = idValue + '.intValue()';
} else if (primaryKeyType === 'UUID') {
    idValue = idValue + '.toString()';
}
```

Comment (gzsombor) [reply generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs]: Good idea!

**Pull request merged**